Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -67,23 +73,29 @@ IRelocator relocator
       return false;
     }
 
-    public IObservable<DafnyDocument> OpenDocument(DocumentTextBuffer document) {
+    public void OpenDocument(DocumentTextBuffer document) {
       var cancellationSource = new CancellationTokenSource();
       var resolvedDocumentTask = OpenAsync(document, cancellationSource.Token);
 
       var translatedDocument = LoadVerificationTasksAsync(resolvedDocumentTask, cancellationSource.Token);
-      var verifiedDocuments = Verify(translatedDocument, VerifyOnOpen, cancellationSource.Token);
 
-      documents.Add(document.Uri, new DocumentEntry(
+      var documentEntry = new DocumentEntry(
         document.Version,
+        document,
         resolvedDocumentTask,
         translatedDocument,
-        verifiedDocuments,
-        cancellationSource
-      ));
-      return resolvedDocumentTask.ToObservableSkipCancelled().Where(d => !d.LoadCanceled).
-        Concat(translatedDocument.ToObservableSkipCancelled()).
-        Concat(verifiedDocuments);
+        cancellationSource,
+        new RequestUpdatesOnUriObserver(logger, telemetryPublisher, notificationPublisher, document)
+      );
+      documents.Add(document.Uri, documentEntry);
+
+      documentEntry.Observe(
+        resolvedDocumentTask.ToObservableSkipCancelled().Where(d => !d.LoadCanceled).
```

Comment:
Would it be possible, perhaps, to not just skip cancelled tasks, but also tasks which contain errors?